modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -726,7 +727,7 @@ uint32_t PngDecoder::read_chunk(Chunk& chunk)
         if (size != 8 + 1 + 4 && size != 8 + 2 + 4 && size != 8 + 6 + 4)
             return 0;
     } else if (id != id_fdAT && id != id_IDAT && id != id_IEND && id != id_PLTE && id != id_tRNS) {
-        if (size > PNG_USER_CHUNK_MALLOC_MAX)
+        if (size > PNG_USER_CHUNK_MALLOC_MAX && id != id_tEXt)
```

Comment:
Please just move id_tExt to the line above and respect the alphabetical order. Thx. This PNG_USER_CHUNK_MALLOC_MAX could also be an env variable that the user can change.

Reply:
> Please just move id_tExt to the line above and respect the alphabetical order. Thx. This PNG_USER_CHUNK_MALLOC_MAX could also be an env variable that the user can change.

done.